### ISSUE TITLE
upcoming mpv release is incompatible due to API change

### ISSUE THREAD

**Issue by wiiaboo** [Syncplay/syncplay#261]
https://github.com/mpv-player/mpv/commit/f57f13ceb031e5a11a4a2de226ee1f9b211ddb64

No date for the removal of `input-file`, but it's finally being deprecated.

**Comment by Et0h** [comment]
Thanks for letting us know @wiiaboo

I have had an initial look at https://github.com/mpv-player/mpv/blob/master/DOCS/man/ipc.rst and it looks like moving to input-ipc-server would require moving to JSON and might also involve having to work around some UTF-8 quirks and might have some additional issues working on Windows, so not a straightforward switch from one command line switch to another.

Ideally we would want any change to be done in a way which maintained backwards compatibility with mpv >= 0.6.2 (which as per https://syncplay.pl/guide/install/ is the earliest version of mpv we say we support).

**Comment by wiiaboo** [comment]
I don't think it's sane to still support that old an mpv. At that point, still supporting mplayer/mplayer2 makes as much sense. Might as well just create a new mpv thing and that should be supported since at least the moment named pipes were supported in Windows (~[v0.17.0](https://github.com/mpv-player/mpv/commit/5bf473d5ca3b1a8f0902d08799b28512eb81070b)).

I've been using a python script to send URLs from the browser to a single-instance mpv for years now, so named pipes should be fine. Not sure about two-way communication though. My script is one-way only:

```python
#!python3
#!/usr/bin/env python3
#mpv-append.pyw
import sys, subprocess, shlex, codecs, os
from time import strftime
import argparse

log_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'umpv.log')

parser = argparse.ArgumentParser(prefix_chars='-/')
parser.add_argument('mpvopts',
                    help='opts passed to mpv if already open',
                    nargs='+')
parsed_args = parser.parse_args(sys.argv[1:])

with codecs.open(log_path, 'a', 'utf-8') as log_file:
    if len(sys.argv[1:]) > len(parsed_args.mpvopts):
        log_file.write('[{}] {} ⇒ "{}"\n'.format(strftime("%Y-%m-%d %H:%M:%S"),
            sys.argv[1:], '" "'.join(parsed_args.mpvopts)))
    else:
        log_file.write('[{}] "{}"\n'.format(strftime("%Y-%m-%d %H:%M:%S"),
            sys.argv[1:], '" "'.join(parsed_args.mpvopts)))

try:
    f = open(r'\\.\pipe\mpvsocket', 'r+b', 0)
    for arg in parsed_args.mpvopts:
        if not arg.startswith('-'):
             subprocess.run(["echo", "loadfile", arg.replace('&','^&'),
                "append-play"], stdout=f, shell=True)
except OSError:
    args = ['mpv.exe', '--profile=umpv', '--input-ipc-server=\\\\.\\pipe\\mpvsocket']
    args += parsed_args.mpvopts
    subprocess.Popen(args)
```

**Comment by daniel-123** [comment]
I wouldn't mind raising minimum required mpv version. Just for perspective - Debian Stable uses 0.29, Oldstable 0.23 and latest Ubuntu LTS 0.27. Most relevant distribution that still uses mpv older than 0.17 is previous Ubuntu LTS which sits at 0.14.

On the other hand 0.17 only [renamed the option --input-unix-socket to --input-ipc-server](https://github.com/mpv-player/mpv/commit/cd50ebba363315e15276e3b0895c33e85813fc4a). Under its prior name [it was introduced all the way back in 0.7](https://github.com/mpv-player/mpv/commit/13039414f5fd00a6b20d2bbad9fd497c6e579d34). Given the above it's probably not particularly hard to keep the support for 0.7+, but I'm pretty sure it's still not worthwhile.

No idea about Windows situation, but I guess it's easier to just tell people there to update mpv there.

**Comment by wiiaboo** [comment]
It was renamed because it stopped requiring POSIX sockets/pipes.

**Comment by wiiaboo** [comment]
It's out on git master and on the next release: https://github.com/mpv-player/mpv/commit/b8daef5d8b3f58481ac95f7217df8f1509fa0b59

**Comment by Et0h** [comment]
Thanks for the notification @wiiaboo - hopefully moving over to JSON won't be too tricky.

**Comment by C0rn3j** [comment]
Syncplay 1.6.4a or git now won't work against git mpv, I guess it's because of this change.

`Syncplay is not compatible with this version of mpv. Please use a different version of mpv (e.g. Git HEAD).`

**Comment by Et0h** [comment]
I've now added a notice to the syncplay.pl homepage advising people to stick with mpv 0.6.2 – 0.32.0 for now.

**Comment by wiiaboo** [comment]
Preferably 0.32.0.

**Comment by daniel-123** [comment]
I've adjusted the issue a bit to reflect its importance and urgency.
Additionally I revisited the mpv versions in latest LTS distros - the oldest one I found is 0.27. Based on this I see absolutely no point in trying to keep compatibility with anything remotely as old as 0.6.
New API under current option naming is available from 0.17 onwards and that IMHO should be the new minimum supported version.
